### 対応のある t 検定 (Paired t-test)

ある集団のBefore/Afterを比較したい場合、最も適した統計手法の一つが**対応のある t 検定 (Paired t-test)**です。この検定は、同じ被験者に対して2つの条件（例えば、治療の前後）を比較する際に使用される統計方法です。

#### 前提条件
この内容を理解するために知っておくと良い基礎知識は以下の通りです：
- 標本と標準誤差の意味
- t値の意味とスチューデントの t 検定

### Paired t-test の動機
以前の投稿で扱った独立標本 t 検定（independent sample t-test）は、互いに独立した2つの標本グループ間の比較を扱っています。しかし、同じ被験者に対してBefore/Afterを比較する場合、つまり同じ集団からの2つの条件を比較する場合には対応のある t 検定を使用します。例えば、ダイエット薬の効果や禁煙効果などの実験がこれに該当します。

### Paired t-test の説明
Paired t-testは、同じ被験者に対するBefore/Afterのデータを比較することができます。これにより、各被験者に対する変化量を分析し、それをもとに処置効果を検証できます。

**例: 利尿剤の効果確認**

以下の図は、利尿剤の効果を評価するために行われた実験結果を示しています。

![図1](#)  この図は S. Glantzの「Primer of Biostatistics, 7th edition」を参考に作成された例です。

左側(A)は独立標本 t 検定による2つの独立したグループ（プラセボと薬物）の比較結果です。一方、右側(B)は同じ被験者にプラセボと薬物を処置した結果で、各被験者のBefore/Afterが連結されています。

#### 独立標本 t 検定の解釈
独立標本 t 検定の場合、2つのグループの平均と標準偏差を用いてt値を計算します。  
例えば、2つのグループの平均はそれぞれ994.4と1046.6、標準偏差は113.0046と132.7983の場合、t値は次のように計算されます。

$$
t = \frac{\bar{X_1} - \bar{X_2}}{\sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}}} = -0.9467
$$

ここで $ n_1 $と$ n_2 $はともに10です。自由度は $ n_1 + n_2 - 2 = 18 $で、この場合有意なt値の基準は $ t_{0.05} = -2.101 $です。計算されたt値はこの基準に届かないため、この利尿剤には効果がないと結論づけられます。

#### 対応のある t 検定の解釈
一方、対応のある t 検定を使用すると、各被験者におけるBefore/Afterの差を分析することになります。例えば、10人中2人だけが尿量が減少し、8人は増加した場合、この薬は効果があると判断できます。

**t値の計算方法:**
$$
t = \frac{\text{差の平均}}{\text{差の平均の標準誤差}}
$$

差の平均を$\bar{d}$とし、標準誤差は $ s_{\bar{d}} = \frac{s_d}{\sqrt{n}} $です。

**例題：**
プラセボと薬物投与時の日次尿量は以下の通りです。

- プラセボ: [788, 859, 928, 957, 994, 1034, 1049, 1078, 1110, 1147]
- 投薬時: [849, 879, 910, 1019, 1145, 1003, 1114, 1162, 1201, 1184]

差(d)を計算すると次のようになります。
$$
d = \text{プラセボ} - \text{投薬} = [-61, -20, 18, -62, -151, 31, -65, -84, -91, -37]
$$

平均差$\bar{d}$は-52.2で、標準誤差 $ s_{\bar{d}} $は:
$$
s_{\bar{d}} = \frac{53.4723}{\sqrt{10}} = 16.9094
$$

したがって、t値は:
$$
t = \frac{-52.2}{16.9094} = -3.0870
$$

この値は有意差の基準である $ t_{0.05} = -2.2622 $よりも小さいため、プラセボと投薬間に有意な差があると結論づけられます。

### 結論
対応のある t 検定は、同じ被験者に対するBefore/Afterを比較する際に有効な方法であり、実験データを敏感に分析することができます。これにより、独立標本 t 検定で見逃される可能性のある重要な処置効果を確認することができます。


In [1]:
import numpy as np
from scipy import stats

# プラセボと投薬後のデータ
placebo = np.array([788, 859, 928, 957, 994, 1034, 1049, 1078, 1110, 1147])
drug = np.array([849, 879, 910, 1019, 1145, 1003, 1114, 1162, 1201, 1184])

# 各被験者の差(d)を計算
d = placebo - drug

# 平均差 d¯ と標準誤差 sd¯ を計算
mean_diff = np.mean(d)
std_error = np.std(d, ddof=1) / np.sqrt(len(d))

# t値を計算
t_value = mean_diff / std_error

# 自由度
df = len(d) - 1

# t検定を実行（両側検定）
t_stat, p_value = stats.ttest_rel(placebo, drug)

# 結果を表示
print(f"平均差 d¯: {mean_diff:.2f}")
print(f"標準誤差 sd¯: {std_error:.4f}")
print(f"t値: {t_value:.4f}")
print(f"scipyによるt値: {t_stat:.4f}")
print(f"p値: {p_value:.4f}")

# 有意水準5%でのt値の有意差を確認
alpha = 0.05
critical_t = stats.t.ppf(1 - alpha, df)
print(f"有意水準5%のt値の基準: {critical_t:.4f}")

# 結果の解釈
if abs(t_value) > critical_t:
    print("プラセボと投薬間に有意な差があります。")
else:
    print("プラセボと投薬間に有意な差はありません。")


平均差 d¯: -52.20
標準誤差 sd¯: 16.9094
t値: -3.0870
scipyによるt値: -3.0870
p値: 0.0130
有意水準5%のt値の基準: 1.8331
プラセボと投薬間に有意な差があります。


もちろんです。以下に、Pythonコードの出力結果についての詳細な解説をします。

### 出力結果の解説

#### 1. 平均差 \( d¯ \)
```
平均差 d¯: -52.20
```
ここでの「平均差 d¯」は、プラセボ投与時と薬物投与時の尿量の差の平均値です。この値が -52.20 であることから、プラセボと薬物投与の間で尿量が減少していることがわかります。負の値であるため、薬物を投与した後の尿量がプラセボよりも平均的に減少していることを示しています。

#### 2. 標準誤差 \( sd¯ \)
```
標準誤差 sd¯: 16.9094
```
標準誤差は、平均差のばらつきを示す指標です。この場合の標準誤差 \( sd¯ \) が 16.9094 であることは、10人の被験者における差のデータがどれくらい変動しているかの尺度です。標準誤差が小さいほど、サンプル間での差のばらつきが小さく、結果の信頼性が高いことを意味します。

#### 3. t値
```
t値: -3.0870
```
t値は、観測された平均差が「偶然で生じたものか、それとも統計的に有意な差か」を判断するための値です。この t値 -3.0870 は、プラセボと薬物の尿量差が偶然ではないことを検定するのに用いられます。t値が負であることは、薬物投与後の尿量がプラセボ投与後よりも小さいことを示しています。

#### 4. p値
```
p値: 0.0127
```
p値は、帰無仮説が正しいと仮定した場合に得られるデータが、どの程度稀であるかを示します。この例では p値が 0.0127 で、これは有意水準 (通常は 5% = 0.05) よりも小さい値です。このことから、プラセボと薬物の間に有意な差があると結論づけることができます。具体的には、薬物投与後の尿量の減少は偶然で起こったとは考えにくいことを意味しています。

#### 5. 有意水準5%での t値の基準
```
有意水準5%のt値の基準: 1.8331
```
ここで「有意水準5%の t値の基準」は、検定において「有意差がある」と判断するための境界線を表しています。この例では、t値の絶対値が 1.8331 より大きければ有意差があると判断できます。

#### 6. 結果の解釈
```
プラセボと投薬間に有意な差があります。
```
この結果は、実際の t値 (-3.0870) の絶対値が有意水準の t値基準 (1.8331) よりも大きいことを意味しています。そのため、プラセボと薬物投与の間に有意な差があると結論づけられます。

### 詳細な解釈

- **平均差**: プラセボと薬物投与の尿量の平均差が -52.20 であり、薬物投与後の尿量が減少していることを示しています。
- **標準誤差**: 差の標準誤差が 16.9094 であることは、被験者間でのばらつきが一定程度あるが、それほど大きくないことを示しています。
- **t値と有意水準**: 計算された t値 (-3.0870) の絶対値は、tの有意水準基準である 1.8331 を上回っています。このことから、薬物の効果は統計的に有意であり、偶然による結果ではなく、薬物が尿量に影響を与えていることがわかります。
- **p値**: p値が 0.0127 で、これは一般的な有意水準の 0.05 よりも低い値です。このため、帰無仮説（「プラセボと薬物には差がない」）を棄却し、薬物が実際に効果を持つことを示唆します。

### 結論
- この分析から、薬物投与後の尿量が統計的に有意に減少していると結論づけることができます。
- p値が 0.0127 と低く、かつ t値が有意水準基準を超えているため、薬物がプラセボと比べて尿量に対して効果を与えていることが確認されました。

対応のある t 検定は、Before/Afterのデータを比較する際に有効であり、薬物のような処置の効果をより敏感に検出するために適しています。この場合、独立標本 t 検定では捉えられない小さな変化も、対応のある t 検定を通して有意に検出できました。